quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/CollectionManagerImpl.java

```diff
@@ -84,8 +93,30 @@ public void deleteCollection(String id) throws CollectionNotFoundException {
             // If it is linked to a network, we will archive the collection.
             collectionRepo.delete(collection);
         } else {
-            throw new CollectionNotFoundException();
+            throw new CollectionNotFoundException("CollectionId: " + id);
+        }
+    }
+    
+    
+    /**
+     * This method checks whether a collection with given collectionId exists and
+     * returns the collection if it exists.
+     * If it doesn't exist, it simply throws an exception.
+     * 
+     * @param collectionId is the id of the collection to be checked
+     * @return the Collection entry found in the database
+     * @throws InvalidObjectIdException    if collectionId couldn't be conveted to
+     *                                     ObjectId
+     * @throws CollectionNotFoundException if collection with given collectionId
+     *                                     does't exist
+     */
+    @Override
+    public Collection getCollection(String collectionId) throws InvalidObjectIdException, CollectionNotFoundException {
+        Collection collection = findCollection(collectionId);
+        if (collection == null) {
+            throw new CollectionNotFoundException("CollectionId: " + collectionId);
         }
+        return collection;
```

Comment:
this method should just return null if the collection doesn't exist. Retrieving an object that doesn't exist is something that could easily be used across the application (simply to check if a collection exist) and exceptions shouldn't be used in such cases (too expensive to keep building a stacktrace and more annoying to incorporate in the code).

Reply:
After a network is submitted for a collection, two scenarios could occur -
1. The collection exists, and it's the first time a network is being submitted to it
2. The collection doesn't exist, i.e. the network submission request is incorrect

In the first case, we add new mappedTripleGroup to the existing collection
In the second case, we return null, which will send a NOT_FOUND response to the request

Now here, if we don't retrieve the collection and simply pass `new ObjectId(collectionId)`, this method will return null for both the above cases
To distinguish between the two, we need to retrieve the collection here

I can change this method to -
Try to get Collection, and if it doesn't exist, throw CollectionNotFoundException here
If it exists, null will be returned, which means we need to add new MappedTripleGroup

Is that fine?

Reply:
Yes that makes sense. In that case I'll just remove this method altogether and just utilize the existing `findCollection` method